crmsh/bootstrap.py

```diff
@@ -228,13 +231,53 @@ def wait_for_cluster():
     status_long("Waiting for cluster")
     while True:
         _rc, out, _err = utils.get_stdout_stderr("crm_mon -1")
-        if "online" in out.lower():
+        if is_online(out):
             break
         status_progress()
-        sleep(5)
+        sleep(2)
     status_done()
 
 
+def get_cluster_node_hostname():
+    """
+    Get the hostname of the cluster node used during the join process if an IP address is used.
+    """
+    peer_node = None
+    if _context.cluster_node:
+        if utils.valid_ip_addr(_context.cluster_node):
+            rc, out, err = utils.get_stdout_stderr("ssh {} hostname".format(_context.cluster_node))
```

Comment:
Since a system might have multiple names corresponding to different IP addresses in /etc/hosts, to make sure we are really getting the `node name` used for the cluster, we'd better run `crm_node --name` on that node but of course while cluster service is running there.

Reply:
And of course, probably depending on how this function will be potentially used, use `uname --n` as a fallback in case `crm_node --name` fails for example if cluster service is not running there yet?

Reply:
Changed as `crm_node --name`, and also changed for crmsh-4.1 and crmsh-3.0 branche PRs.

I think if the cluster service is really stopped on init node while executing this piece of code, just let `error` to terminate the join process, because in that case, we should stop and check why service on init node stopped, instead of going ahead.

Thanks @gao-yan !